FastNoiseSIMD/FastNoiseSIMD_internal.cpp

```diff
@@ -190,7 +191,7 @@ static SIMDf VECTORCALL FUNC(FLOOR)(SIMDf a)
 {
 	SIMDf fval = SIMDf_CONVERT_TO_FLOAT(SIMDi_CONVERT_TO_INT(a));
 
-	return vsubq_f32(fval, SIMDf_AND(SIMDf_LESS_THAN(a, fval), SIMDf_NUM(1)));
+	return vsubq_f32(fval, SIMDf_CAST_TO_FLOAT(vandq_s32(SIMDf_LESS_THAN(a, fval), SIMDi_NUM(1))));
```

Comment:
I'm only unsure whether this is right; the inputs should be int, but I'm not sure if `SIMDi_NUM` does the right thing.

Reply:
This won't work, it is meant to conditionally subtract 1 if a < fval. You are subtracting an int 1 instead of a float 1 these are bit casted not converted

Reply:
OK, that's what I thought; I think it should be correct now.

Reply:
You shouldn't need to change the floor function. See my latest commit

Reply:
You haven't pushed any new commits.

Reply:
My bad, looks like our changes are the same anyway, apart from the floor function

Reply:
The change to `floor` is still necessary; on `master`:
```
In file included from FastNoiseSIMD/FastNoiseSIMD_neon.cpp:37:0:
FastNoiseSIMD/FastNoiseSIMD_internal.cpp: In function ‘SIMDf L5_FUNC_FLOOR(SIMDf)’:
FastNoiseSIMD/FastNoiseSIMD_internal.cpp:184:77: error: cannot convert ‘int32x4_t {aka __vector(4) int}’ to ‘float32x4_t {aka __vector(4) float}’ for argument ‘1’ to ‘int32x4_t vreinterpretq_s32_f32(float32x4_t)’
 #define SIMDf_AND(a,b) SIMDf_CAST_TO_FLOAT(vandq_s32(vreinterpretq_s32_f32(a),vreinterpretq_s32_f32(b)))
                                                                             ^
FastNoiseSIMD/FastNoiseSIMD_internal.cpp:154:54: note: in definition of macro ‘SIMDf_CAST_TO_FLOAT’
 #define SIMDf_CAST_TO_FLOAT(a) vreinterpretq_f32_s32(a)
                                                      ^
FastNoiseSIMD/FastNoiseSIMD_internal.cpp:193:25: note: in expansion of macro ‘SIMDf_AND’
  return vsubq_f32(fval, SIMDf_AND(SIMDf_LESS_THAN(a, fval), SIMDf_NUM(1)));
                         ^~~~~~~~~
```